google-chart-loader.html

```diff
@@ -0,0 +1,206 @@
+<link rel="import" href="../polymer/polymer.html">
+<link rel="import" href="../promise-polyfill/promise-polyfill-lite.html">
+<link rel="import" href="charts-loader.html">
+
+<script>
+(function() {
+  "use strict";
+
+  // Most charts use this package.
+  var DEFACTO_CHART_PACKAGE = 'corechart';
+
+  // A collection of chart types associated with a function that returns a
+  // Promise for the chart type's constructor.
+  var CHART_CONSTRUCTORS = {
+    'area': {
+      ctor: 'AreaChart',
+    },
+    'bar': {
+      ctor: 'BarChart',
+    },
+    'md-bar': {
+      ctor: 'Bar',
+      pkg: 'bar',
+    },
+    'bubble': {
+      ctor: 'BubbleChart',
+    },
+    'candlestick': {
+      ctor: 'CandlestickChart',
+    },
+    'column': {
+      ctor: 'ColumnChart',
+    },
+    'combo': {
+      ctor: 'ComboChart',
+    },
+    'geo': {
+      ctor: 'GeoChart',
+    },
+    'histogram': {
+      ctor: 'Histogram',
+    },
+    'line': {
+      ctor: 'LineChart',
+    },
+    'pie': {
+      ctor: 'PieChart',
+    },
+    'scatter': {
+      ctor: 'ScatterChart',
+    },
+    'stepped-area': {
+      ctor: 'SteppedAreaChart',
+    },
+    'table': {
+      ctor: 'Table',
+      pkg: 'table',
+    },
+    'timeline': {
+      ctor: 'Timeline',
+      pkg: 'timeline',
+    },
+    'gauge': {
+      ctor: 'Gauge',
+      pkg: 'gauge',
+    },
+    'treemap': {
+      ctor: 'TreeMap',
+      pkg: 'treemap',
+    },
+  };
+
+  function namespaceForType(type) {
+    return google[type.startsWith('md-') ? 'charts' : 'visualization'];
```

Comment:
I think we may be talking about different things.
In this case, the namespace is where the chart constructor is located.
For traditional chart types, that namespace is `google.visualization`.
For new (material) charts, that namespace is `google.charts`.

e.g.
google.visualization.BarChart // traditional bar chart
google.charts.Bar  // material bar/column chart

So there is no default that would work. The calling function would look for `defaultns.BarChart` or something which would be equally incorrect.

This function is called inside of a protected area where the required package is guaranteed to have already been loaded. There is no way to recover from a situation where the package is loaded but `google` or `google.charts` is not available.

The exception thrown `Uncaught ReferenceError: google is not defined` LGTM in this case. If we were to check if google exists, it would just be to throw a different exception.

What do you think?